datamap/mv/plan/src/main/scala/org/apache/carbondata/mv/plans/util/BirdcageOptimizer.scala

```diff
@@ -156,6 +159,53 @@ object BirdcageOptimizer extends RuleExecutor[LogicalPlan] {
   }
 }
 
+// This class is used to optimize the filter condition
+// If all of the conditions have alias on the aggregate expressions
+// it will be pull up and exchange the alias to match the MV
+object MVPredicateAlias extends Rule[LogicalPlan] with PredicateHelper {
```

Comment:
Whatever is pushdown in `PushDownPredicate` you are trying to pull up using this rule. Please just remove the `PushDownPredicate` from the optimizer will solve all having clause issues.  But the only issue will be with filter query on groupby queries , I think that issue can be fixed in MV module instead of adding a rule.